src/app/InteractionModelEngine.cpp

```diff
@@ -30,10 +30,16 @@
 #include "access/RequestPath.h"
 #include "access/SubjectDescriptor.h"
 #include <app/RequiredPrivilege.h>
+#include <app/util/af-types.h>
 #include <lib/core/TLVUtilities.h>
 #include <lib/support/CodeUtils.h>
 
 extern bool emberAfContainsAttribute(chip::EndpointId endpoint, chip::ClusterId clusterId, chip::AttributeId attributeId);
+extern const EmberAfEndpointType * emberAfFindEndpointType(chip::EndpointId endpointId);
```

Comment:
should we have these in some header somewhere?

I see we did it once ... but now we declare 6 externs and this starts being ugly. This is why headers exist.

Reply:
This maybe bypasses dependency/layering  ... however circular dependency rules should be useful.

Reply:
Yes, but the relevant headers pull in generated headers....  I will try to de-tangle that, but it will take some work.

Filed https://github.com/project-chip/connectedhomeip/issues/26764